src/cert_context.rs

```diff
@@ -107,6 +121,69 @@ impl CertContext {
         self.get_encoded_bytes()
     }
 
+    /// Deletes the cert's associated key container for both CAPI and CNG keys
+    pub fn delete_key_container(&mut self) -> io::Result<()> {
+        unsafe {
+            // get CERT_KEY_PROV_INFO_PROP_ID and cast to key_prov
+            let bytes = self.get_bytes(Cryptography::CERT_KEY_PROV_INFO_PROP_ID)?;
+            assert!(bytes.len() <= u32::max_value() as usize);
+            let key_prov = bytes.as_ptr() as *const Cryptography::CRYPT_KEY_PROV_INFO;
+            
+            if (*key_prov).dwKeySpec == 0 { // CNG Key
+                let mut prov_handle = Cryptography::NCRYPT_PROV_HANDLE::default();
+                let ncrypt_prov:NcryptProvHandle;
+                let mut key_handle = Cryptography::HCRYPTPROV_OR_NCRYPT_KEY_HANDLE::default();
+                
+                let ret = Cryptography::NCryptOpenStorageProvider(&mut prov_handle, (*key_prov).pwszProvName, 0);
+                if ret == 0 {
+                    ncrypt_prov = NcryptProvHandle::from_inner(prov_handle);
+                } else {
+                    return Err(io::Error::from_raw_os_error(ret));
+                }
```

Comment:
A more idiomatic way to express this is:

```rust
let ncrypt_prov = match Cryptography::NCryptOpenStorageProvider(&mut prov_handle, (*key_prov).pwszProvName, 0) {
    0 => NcryptProvHandle::from_inner(prov_handle),
    e => return Err(io::Error::from_raw_os_error(e)),
};
```

though I'd probably factor this out into a function too